src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+// write code here;
+
+export function groupByKey<T>(arrayOfObjects: T[], key: keyof T): GroupsMap<T> {
+  const newArrayOfObjects: GroupsMap<T> = {};
+
+  arrayOfObjects.map((objectInArray) => {
+    const keyValue = String(objectInArray[key]);
+
+    if (!newArrayOfObjects[keyValue]) {
+      newArrayOfObjects[keyValue] = [];
+    }
+
+    newArrayOfObjects[keyValue].push(objectInArray);
+  });
```

Comment:
The `map` function is being used here, but the return value is not utilized. Since the purpose is to iterate over the array and not to transform it, `forEach` would be a more appropriate choice. This will avoid any confusion about the intent of the code.